avro4s/src/main/scala/com/ovoenergy/kafka/serialization/avro4s/Avro4sSerialization.scala

```diff
@@ -57,30 +57,38 @@ private[avro4s] trait Avro4sSerialization {
                                                     isKey: Boolean,
                                                     includesFormatByte: Boolean): KafkaDeserializer[T] = {
     val schemaRegistryClient = JerseySchemaRegistryClient(schemaRegistryClientSettings)
-    avroBinarySchemaIdDeserializer(schemaRegistryClient, isKey, () => schemaRegistryClient.close(), includesFormatByte)
+    avroBinarySchemaIdDeserializerWithProps(schemaRegistryClient, isKey, () => schemaRegistryClient.close(), includesFormatByte)
   }
 
   /**
     * Build a deserializer for binary-encoded messages prepended with a four byte Schema Registry schema ID.
     *
     * @param isKey true if this is a deserializer for keys, false if it is for values
     * @param includesFormatByte whether the messages are also prepended with a magic byte to specify the Avro format
+    * @param props configuration of the underlying [[KafkaAvroDeserializer]]
     */
   def avroBinarySchemaIdDeserializer[T: FromRecord](schemaRegistryClient: SchemaRegistryClient,
                                                     isKey: Boolean,
-                                                    includesFormatByte: Boolean): KafkaDeserializer[T] =
-    avroBinarySchemaIdDeserializer(schemaRegistryClient, isKey, () => Unit, includesFormatByte)
+                                                    includesFormatByte: Boolean,
+                                                    props: Map[String, String] = Map()): KafkaDeserializer[T] =
+    avroBinarySchemaIdDeserializerWithProps(schemaRegistryClient, isKey, () => Unit, includesFormatByte, props)
```

Comment:
I intended to add the parameter with default: `props: Map[String, String] = Map()` to all related methods, however the Scala compiler won't allow this. See [this StackOverflow](https://stackoverflow.com/questions/4652095/why-does-the-scala-compiler-disallow-overloaded-methods-with-default-arguments).

I opted to add it to the overloads which take the `SchemaRegistryClient` which seemed best.